src/treeview_sandbox.ml

```diff
@@ -79,6 +79,134 @@ module Command = struct
+            let message =
+              "\"OCaml: Generate Documentation\" command only works with OPAM \
+               sandboxes."
+            in
+            let (_ : 'a option Promise.t) =
+              Window.showErrorMessage ~message ()
+            in
+            Promise.resolve ()
+          | Sandbox.Opam (opam, switch) -> (
+            let* odig = Odig.of_opam (opam, switch) in
+            match odig with
+            | Error e ->
+              let message =
+                match e with
+                | Odig.Odig_not_installed ->
+                  "\"OCaml: Generate Documentation\": the package \"odig\" \
+                   must be installed to generate documentation."
+              in
+              let (_ : 'a option Promise.t) =
+                Window.showErrorMessage ~message ()
+              in
+              Promise.resolve ()
+            | Ok odig -> (
+              let is_live_preview_extension_installed =
+                Extensions.is_extension_installed "ms-vscode.live-server"
+              in
+              if not is_live_preview_extension_installed then
+                let+ choices =
+                  Window.showErrorMessage
+                    ~message:
+                      "Live preview extension must be installed to show the \
+                       documentation."
+                    ~choices:[ ("Install", `Install) ]
+                    ()
+                in
+                match choices with
+                | None -> ()
+                | Some `Install ->
+                  let (_ : Ojs.t option Promise.t) =
+                    Vscode.Commands.executeCommand
+                      ~command:"workbench.extensions.search"
+                      ~args:[ Ojs.string_to_js "ms-vscode.live-server" ]
+                  in
+                  ()
+              else
+                let arg = List.hd_exn args in
+                let dep = Dependency.t_of_js arg in
+                let name = Sandbox.Package.name dep in
+                let options =
+                  ProgressOptions.create
+                    ~location:(`ProgressLocation Notification)
+                    ~title:
+                      (Printf.sprintf "Generating documentation for %s" name)
+                    ~cancellable:false ()
+                in
+                let task ~progress:_ ~token:_ = Odig.odoc_exec odig name in
+                let* result =
+                  Vscode.Window.withProgress
+                    (module Interop.Js.Result
+                              (Interop.Js.String)
+                              (Interop.Js.String))
+                    ~options ~task
+                in
+                match result with
+                | Error _ ->
+                  let (_ : 'a option Promise.t) =
+                    Window.showErrorMessage
+                      ~message:
+                        (Printf.sprintf
+                           "Error while generating documentation for %s" name)
```

Comment:
Can this error occur in practice? If so, how can the user find out more about the cause?

Reply:
Yes, that can happen for instance with packages that don't have any documentation to generate. An example is `ocaml-lsp-server` which shows up in the list of packages installed in the current switch.

When you try to generate the documentation for that package `odig odoc ocaml-lsp-server`, the command actually succeed but no documentation is being written to disk:

```bash
~/Dev/vscode-ocaml-platform offline-doc ≡ 1.55s
❯ odig odoc ocaml-lsp-server
Updating documentation, this may take some time...

~/Dev/vscode-ocaml-platform offline-doc ≡ 1.16s
❯ echo $status
0
```

Hence why we [verify that the expected documentation folder for the package was created and return an error if not](https://github.com/ocamllabs/vscode-ocaml-platform/pull/771/files#diff-dee540601fb09da08934f2e16441a5ef20ef7a1eaf925e55a200b9f369a02aaaR45).

The interesting thing is that while the `odoc` command doesn't produce any error, the `doc` command behave differently:

```bash
❯ odig doc ocaml-lsp-server
Updating documentation, this may take some time...
odig: [ERROR] No doc could be generated for:
              '/Users/corentin/Dev/vscode-ocaml-platform/_opam/var/cache/odig/html/ocaml-lsp-server/index.html'

~/Dev/vscode-ocaml-platform offline-doc ≡ 0.81s
❯ echo $status
3
```

I thought that was an issue on `odig` side but that's the expected behaviour. See relevant issue [here](https://github.com/b0-system/odig/issues/64)